File: compiled_lang/cpp/src/f03.loop.cpp
// testbox: title="indexed for loop over std::vector"
#include <algorithm>
#include <filesystem>
#include <iostream>
#include <string>
#include <vector>

namespace fs = std::filesystem;

int main() {
    std::vector<std::string> items;
    for (const auto& entry : fs::directory_iterator("dirtest"))
        items.push_back(entry.path().filename().string());
    std::sort(items.begin(), items.end());

    // indexed for loop
    for (std::size_t i = 0; i < items.size(); ++i) {
        const std::string& item = items[i];
        if (fs::is_directory("dirtest/" + item))
            std::cout << item << " is a directory" << std::endl;
        else
            std::cout << item << " is not a directory" << std::endl;
    }

    return 0;
}

File: compiled_lang/cs/src/f03.loop.cs
// testbox: title="indexed for loop over List<T>"
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;

class F03Loop
{
    static void Main()
    {
        List<string> items = Directory.GetFileSystemEntries("dirtest")
                                       .Select(Path.GetFileName)
                                       .OrderBy(name => name)
                                       .ToList();

        for (int i = 0; i < items.Count; i++)
        {
            string item = items[i];
            if (Directory.Exists(Path.Combine("dirtest", item)))
                Console.WriteLine($"{item} is a directory");
            else
                Console.WriteLine($"{item} is not a directory");
        }
    }
}
